src/VirtoCommerce.QuoteModule.Web/Controllers/Api/QuoteModuleController.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using VirtoCommerce.CartModule.Core.Model;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.QuoteModule.Core.Models;
+using VirtoCommerce.QuoteModule.Core.Services;
+using VirtoCommerce.ShippingModule.Core.Model.Search;
+using VirtoCommerce.ShippingModule.Core.Services;
+using VirtoCommerce.StoreModule.Core.Services;
+
+namespace VirtoCommerce.QuoteModule.Web.Controllers.Api
+{
+    [Route("api/quote/requests")]
+    [Authorize(Core.ModuleConstants.Security.Permissions.Read)]
+    public class QuoteModuleController : Controller
+    {
+        private readonly IQuoteRequestService _quoteRequestService;
+        private readonly IQuoteTotalsCalculator _totalsCalculator;
+        private readonly IStoreService _storeService;
+        private readonly IShippingMethodsSearchService _shippingMethodsSearchService;
+
+        public QuoteModuleController(IQuoteRequestService quoteRequestService, IQuoteTotalsCalculator totalsCalculator, IStoreService storeService, IShippingMethodsSearchService shippingMethodsSearchService)
+        {
+            _quoteRequestService = quoteRequestService;
+            _totalsCalculator = totalsCalculator;
+            _storeService = storeService;
+            _shippingMethodsSearchService = shippingMethodsSearchService;
+        }
+
+        /// <summary>
+        /// Search RFQ by given criteria
+        /// </summary>
+        /// <param name="criteria">criteria</param>
+        [HttpPost]
+        [Route("search")]
+        public async Task<ActionResult<QuoteRequestSearchResult>> Search([FromBody]QuoteRequestSearchCriteria criteria)
+        {
+            var retVal = await _quoteRequestService.SearchAsync(criteria);
+            return Ok(retVal);
+        }
+
+        /// <summary>
+        /// Get RFQ by id
+        /// </summary>
+        /// <remarks>Return a single RFQ</remarks>
+        /// <param name="id">RFQ id</param>
+        [HttpGet]
+        [Route("{id}")]
+        public async Task<ActionResult<QuoteRequest>> GetById(string id)
```

Comment:
If **NotFound** result should not be interpreted as exceptional situation by client, please annonate this action method with 
`[ProducesResponseType(typeof(void), StatusCodes.Status404NotFound)]`

Reply:
ok